### ISSUE TITLE
Failed download for pictures ending with .jpg:large (Twitter)

### ISSUE THREAD

**Issue by 3ship** [mcdamo/tab-image-saver#48]
When pictures with the suffix `:large` are opened in separate tabs, the extension doesn't recognize them as pictures. This URL format is used for images uploaded to Twitter and shows them in their original dimensions. Removing the suffix throws out a compressed image with a max. size of 1200px

**Comment by mcdamo** [comment]
I do not see any problem saving twitter images, whether `:large`, `:orig` or otherwise. Please give example links where this is failing for you.

**Comment by NNBR1** [comment]
Literally any picture link, with :large, or with :orig. I disabled all my addons, and the problem persists.

The URL structure is https://pbs.twimg.com/media/[filename].jpg:orig. Any URL using this will fail consistently. FF72.0.2 on Win10 x64.

**Comment by mcdamo** [comment]
I cannot replicate this using current tab-image-saver v2.5.6 with default addon settings.

*.jpg:large or *.jpg:orig images are saved with *.jpg filenames.

**Comment by NNBR1** [comment]
I am using 2.5.6 with default settings (delete/reinstall addon). Immediately, I get a "Failed downloads: 1".

I was using this random picture as a test, refreshing all tabs inbetween attempts. https://pbs.twimg.com/media/EQdGtfmWsAM6KwW.jpg:orig
Any other non-twitter picture seems to work. Same behaviour in a clean profile.

![Anotación 2020-02-11 030648](https://user-images.githubusercontent.com/53584202/74206431-b9326f00-4c7b-11ea-9c61-a9a61dd25a0a.png)
![Anotación 2020-02-11 030632](https://user-images.githubusercontent.com/53584202/74206434-bafc3280-4c7b-11ea-8948-8af03460e2c8.png)

What can I do to provide you a log or see any kind of error, or anything I could do to give some sort of report? I'm unable to find what can I do to give any more info on this error.

**Comment by mcdamo** [comment]
@NNBR1 the debug console might have some useful information.

1. Go to **Firefox** > **Addons** menu
2. Next to **Manage Your Extensions** click the _Cog_ button and select **Debug addons**
3. Scroll down to _Tab Image Saver_ and click **Inspect**
This will open the debug console.

Now run the addon on your image and the output might be something like:
```
> GET https://pbs.twimg.com/media/EQdGtfmWsAM6KwW.jpg:orig   [HTTP/1.1 200 Connection established 6208ms]

imagesFailed > Map { "https://pbs.twimg.com/media/EQdGtfmWsAM6KwW.jpg:orig" → "EQdGtfmWsAM6KwW.jpg" }
```
If the _GET_ did not succeed then there may be more information by expanding it down.

**Comment by eduard93** [comment]
Got the same issue, turned out it's because of the tracking protection:
![image](https://user-images.githubusercontent.com/5127457/74414234-0e27de00-4e52-11ea-97aa-220b3fbfb555.png)

Disabled tracking protection:
![image](https://user-images.githubusercontent.com/5127457/74414389-66f77680-4e52-11ea-89fa-4834e93b4300.png)

And the extension now works.

**Comment by mcdamo** [comment]
Yes, setting tracking protection to Strict I can get this error.

The error returned to the extension appears as general network failure so it can't detect when this issue is the cause of download failures.

I will add some info to the extension for users reduce or disable the level of tracking protection.

**Comment by mcdamo** [comment]
Unfortunately it appears Firefox is not interested in providing any solution for this, other extensions are having the same problem:
- [Tracking protection should not apply on extension pages](https://bugzilla.mozilla.org/show_bug.cgi?id=1404610) (WONTFIX)
- [HTTP requests from WebExtensions to URLs should not be blocked by tracking protection](https://bugzilla.mozilla.org/show_bug.cgi?id=1514054) (WONTFIX)

**Comment by NNBR1** [comment]
Weeeeell. At least the culprit has been found, and if it's up to Firefox to (not) solve, then I guess case closed. In case someone else finds this error, maybe a warning on the configuration page of the addon, like "try disabling tracking protection if some downloads, like Twitter, fail" would be fine, and that's about as much as can be done about it.
